src/index.js

```diff
@@ -32,77 +30,69 @@ api.get = (token, options, getCallback) => {
 	const github = githubFactory(token);
 
 	// Get all repositories
-	github.repos
-		.list({type: 'public'})
-		.then((repos) => {
-			// Keep only forks
-			let forks = repos.filter(({fork}) => fork);
+	try {
+		const repos = await github.paginate(github.repos.listForAuthenticatedUser, {
+			type: 'public'
+		});
+		// Keep only forks
+		let forks = repos.filter(({fork}) => fork);
 
-			// Keep only forks owned by a specified user
-			if (options.user) {
-				forks = forks.filter(({owner}) => owner.login === options.user);
-			}
+		// Keep only forks owned by a specified user
+		if (options.user) {
+			forks = forks.filter(({owner}) => owner.login === options.user);
+		}
 
-			let countDoneForks = 0;
+		let countDoneForks = 0;
+		options.progress({
+			countInspected: 0,
+			totalToInspect: forks.length
+		});
+		const forkDone = (fork) => {
+			countDoneForks += 1;
 			options.progress({
-				countInspected: 0,
+				countInspected: countDoneForks,
+				lastInspected: fork.name,
 				totalToInspect: forks.length
 			});
-			const forkDone = (fork) => {
-				countDoneForks += 1;
-				options.progress({
-					countInspected: countDoneForks,
-					lastInspected: fork.name,
-					totalToInspect: forks.length
-				});
-			};
-
-			// Keep only useless forks
-			async.filter(
-				forks,
-				(fork, filterCallback) => {
-					shouldDeleteFork(github, fork, (error, result) => {
-						if (error) {
-							options.warnings(
-								`Failed to inspect ${fork.name}, skipping`,
-								error
-							);
-							result = false;
-						}
-
-						forkDone(fork);
-						filterCallback(null, result);
-					});
-				},
-				(error, forksToDelete) => {
-					if (error) {
-						return getCallback(error);
-					}
+		};
 
+		// Keep only useless forks
+		const processedForks = forks.map(async (fork) => {
+			try {
+				if (await shouldDeleteFork(github, fork)) {
 					// Map to our simple objects
-					const response = forksToDelete.map((fork) => ({
+					return {
 						owner: fork.owner.login,
 						repo: fork.name,
 						url: fork.html_url
-					}));
-					getCallback(null, response);
+					};
 				}
-			);
-		})
-		.then(null, (error) => {
-			getCallback(error);
+			} catch (error) {
+				options.warnings(`Failed to inspect ${fork.name}, skipping`, error);
+			} finally {
+				forkDone(fork);
+			}
 		});
+
+		const forksToDelete = await Promise.all(processedForks);
+		getCallback(null, forksToDelete.filter(Boolean));
+	} catch (error) {
+		getCallback(error);
+	}
```

Comment:
For public functions, when we make them async, it creates an expectations that they can now be used as promises. As is, this will be promises that do not return correct values, that’ll be very confusing. Instead, let’s return the values, and we’ll modernize the API, too!

```suggestion
		const forksToDelete = (await Promise.all(processedForks)).filter(Boolean);
		getCallback(null, forksToDelete);
		return forksToDelete;
	} catch (error) {
		getCallback(error);
		throw error;
	}
```

Reply:
If this is all that’s needed that’s fine, but if it requires more changes to other parts I’d leave it for a separate PR, perhaps to completely drop the callback. 

To resolve the issue at hand I can use:

``` js 
api.get = (callback) => {
  realApiGet().then(() => callback(null, foo), callback)
}
``` 

So the original api is unaffected.

Reply:
You’re right about the uncaught exception. I was planning to release a new major version anyway, and we could have the promises be the primary support version.

If you’d prefer smaller steps, which I also like, then your suggested workaround is great.

The current situation where we return an incorrect promise is the one that I see trouble with.

Reply:
I saw you prefer smaller steps too so I made the async function an internal one, let me know if the name `_apiGet` is ok.

Now `_apiGet` and `_apiRemove` are pure Promised functions, which you can expose in the next version as is.